### ISSUE TITLE
Misbehaving with mailchimp

### ISSUE THREAD

**Issue by maziarz** [LancerComet/vue-jsonp#7]
Really happy about this lib. Unfortunately it is misbehaving with mailchimp subscriptions for unknown reasons i am trying to figure out.

The code i am using is quite simple:

I have censored parts for reasons.

```
      var self = this
      this.$jsonp('//XXXX.list-manage.com/subscribe/post-json?u=XXXX&id=XXXX', {
        'EMAIL': self.email,
        'FNAME': self.first_name,
        'LNAME': self.last_name
      }).then(json => {
        console.log('success')
        console.log(json)
      }).catch(err => {
        console.log('fail')
        console.log(err)
      })
```

The error i get is following:

```
Uncaught SyntaxError: Unexpected token :
http://XXXX.list-manage.com/subscribe/post-json?u=XXXX&id=XXXX&EMAIL=undefined&FNAME=undefined&LNAME=undefined&callback=jsonp_1745155b8b651a0:1
```

I am not sure which `:` it is referring to, but i can see one in the end of the URI, which may be related?

**Comment by LancerComet** [comment]
Hope you post all codes if possible😙
I have tried in this way and nothing seems to be wrong (v0.1.7):

```javascript
var self = {
  email: 'aa@hotmail.com',
  first_name: 'fn',
  last_name: 'ln'
}

vue.jsonp('//XXXX.list-manage.com/subscribe/post-json?u=XXXX&id=XXXX', {
  'EMAIL': self.email,
  'FNAME': self.first_name,
  'LNAME': self.last_name
}).then(json => {
  console.log('success')
  console.log(json)
}).catch(err => {
  console.log('fail')
  console.log(err)
})
```

And this is what I got:

```javascript
//xxxx.list-manage.com/subscribe/post-json?u=XXXX&id=XXXX&EMAIL=aa%40hotmail.com&FNAME=fn&LNAME=ln&callback=jsonp_20f17f7473ba9a0
```
I think it's something about `self` but I don't know what is that so far.

**Comment by maziarz** [comment]
Sorry. There was a typo, `var self = this`. Everything is passed correctly. I can even see in chrome tools that i am indeed receiving a correct json response.

Chrome version: 62.0.3202.75 (OS X)
vue-jsonp: 0.1.7

I am using nuxt.js

**Comment by LancerComet** [comment]
Cheers! If there is anything I can help just let me know 🍻

**Comment by maziarz** [comment]
I think you misunderstood my message, its still throwing the error with `:`. Sorry :-(

**Comment by LancerComet** [comment]
😂What a shame, I'm going to have a try with nuxt then

**Comment by maziarz** [comment]
Haha, i am so sorry.

**Comment by maziarz** [comment]
So this is interesting. I just ran the site in Firefox with the `build` and got a completely different error: `SyntaxError: missing ; before statement  post-json:1:9`

Linking me to: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Missing_semicolon_before_statement?utm_source=mozilla&utm_medium=firefox-console-errors&utm_campaign=default

**Comment by maziarz** [comment]
Again, the response from Mailchimp seems to look fine to me.

**Comment by maziarz** [comment]
`{"result":"error","msg":"maziar@me.com is already subscribed to list Newsletter Subscribers. <a href=\"https:\/\/XXXX.us14.list-manage.com\/subscribe\/send-email?u=XXXX&id=XXXX&e=XXXX==\">Click here to update your profile.<\/a>"}`

**Comment by LancerComet** [comment]
I'm so sorry but I didn't get any problem, even in Nuxt.js, both dev-env and prod-env 😔

Would you mind me to see all of your related code? I want to know everything about `self`, I think the value of it would be something... strange.

And I don't know whether it is caused by MailChimp because I just know nothing about it. Is MailChimp the kind of thing that you push the code onto it? Does mailchimp modify any code?
This is so weird.

**Comment by maziarz** [comment]
I am sorry i never replied on the full code, here it is. :-)

```
  methods: {
    submitForm: function () {
      var self = this
      this.$jsonp('//rokoko.us14.list-manage.com/subscribe/post-json?u=5453c45de5dad8de9a910ff01&id=b219fac101', {
        'EMAIL': self.email,
        'FNAME': self.first_name,
        'LNAME': self.last_name
      }).then(json => {
        console.log('success')
        console.log(json)
      }).catch(err => {
        console.log('fail')
        console.log(err)
      })
    }
}
```

**Comment by huabingtao** [comment]
流啤

**Comment by huabingtao** [comment]
流啤

**Comment by LancerComet** [comment]
Sorry for being so late, this problem probably is solved in #8
